src/entities/Task/model/slice/taskSlice.test.ts

```diff
@@ -0,0 +1,134 @@
+import { DeepPartial } from '@reduxjs/toolkit';
+import { testSubtask, testTask } from 'shared/test/TestTask';
+import { generateRandomId } from 'shared/lib/helpers';
+import { Status, Task, TasksSchema } from '../types/task';
+import { tasksActions, tasksReducer } from './tasksSlice';
+
+describe('tasksSlice', () => {
+    const state: DeepPartial<TasksSchema> = {
+        data: [
+            testTask,
+        ],
+    };
+    const randomID = generateRandomId();
+    const task: Task = { ...testTask, id: randomID };
+
+    test('Get all todos', () => {
+        const state = {};
+        localStorage.setItem('tasks', JSON.stringify([testTask]));
+        expect(tasksReducer(state as TasksSchema, tasksActions.getAllTasks()))
+            .toEqual({ data: [testTask] });
+    });
+
+    test('Create new task', () => {
+        expect(tasksReducer(state as TasksSchema, tasksActions.createNewTask(task)))
+            .toEqual({
+                data: [
+                    testTask,
+                    { ...testTask, id: randomID },
+                ],
+            });
+    });
+
+    test('Change task status to In Progress', () => {
+        expect(tasksReducer(state as TasksSchema, tasksActions.changeTaskStatus(
+            { taskID: testTask.id, status: Status.IN_PROGRESS },
+        )))
+            .toEqual({
+                data: [
+                    { ...testTask, status: Status.IN_PROGRESS },
+                ],
+            });
+    });
+
+    test('Change task status to Complete', () => {
+        expect(tasksReducer(state as TasksSchema, tasksActions.changeTaskStatus(
+            { taskID: testTask.id, status: Status.COMPLETED },
+        )))
+            .toEqual({
+                data: [
+                    {
+                        id: 'Asdager123',
+                        status: Status.COMPLETED,
+                        completedAt: expect.any(Number),
+                        title: 'test task',
+                        createdAt: 1699860692936,
+                        subTasks: [
+                            {
+                                id: 'bdkldfnb123',
+                                status: Status.COMPLETED,
+                                title: 'test subtask 1',
+                                completedAt: expect.any(Number),
+                            },
+                        ],
+                    },
+                ],
+            });
+    });
```

Comment:
too much copy-pase, update here and other tests to something like this:
```
    test('Change task status to Complete', () => {
        expect(
            tasksReducer(
                state as TasksSchema,
                tasksActions.changeTaskStatus({
                    taskID: testTask.id,
                    status: Status.COMPLETED,
                })
            )
        ).toEqual({
            data: [
                {
                    ...testTask,
                    status: Status.COMPLETED,
                    completedAt: expect.any(Number),
                    subTasks: testTask.subTasks.map((t) => ({
                        ...t,
                        status: Status.COMPLETED,
                        completedAt: expect.any(Number),
                    })),
                },
            ],
        });
    });
```

Reply:
Done